Source/Preferences.swift

```diff
@@ -401,3 +404,38 @@ class Preferences: NSObject {
     @UserDefault(key: kControlEnterOutputKey, defaultValue: 0)
     @objc static var controlEnterOutput: Int
 }
+
+@objc class UserPhraseLocationHelper: NSObject {
+    @objc static var defaultUserPhraseLocation: String {
+        let paths = NSSearchPathForDirectoriesInDomains(.applicationSupportDirectory, .userDomainMask, true)
+        let appSupportPath = paths.first!
+        return (appSupportPath as NSString).appendingPathComponent("McBopomofo")
+    }
+}
+
+extension NSNotification.Name {
+    static var useCustomUserPhraseLocationDidChange = NSNotification.Name(rawValue: "PreferencesUseCustomUserPhraseLocationDidChangeNotification")
+    static var customUserPhraseLocationDidChange = NSNotification.Name(rawValue: "PreferencesCustomUserPhraseLocationNotification")
+}
+
+extension Preferences {
+
+    @UserDefault(key: kUseCustomUserPhraseLocation, defaultValue: false)
+    @objc static var useCustomUserPhraseLocation: Bool {
+        didSet {
+            if useCustomUserPhraseLocation == false {
+                customUserPhraseLocation = nil
+            } else if customUserPhraseLocation == nil || customUserPhraseLocation == "" {
+                customUserPhraseLocation = UserPhraseLocationHelper.defaultUserPhraseLocation
+            }
+            NotificationCenter.default.post(name: .useCustomUserPhraseLocationDidChange, object: useCustomUserPhraseLocation)
```

Comment:
If all the code paths will lead to `.customUserPhraseLocationDidChange` being posted, do we still need `.useCustomUserPhraseLocationDidChange`? Also only the formet gets observed in AppDelegate.